### ISSUE TITLE
On Steam Deck, any screen with a text input opens the on-screen keyboard

### ISSUE THREAD

**Issue by mtolly** [ppy/osu#22190]
### Type

Game behaviour

### Bug description

When playing on the Steam Deck in game mode, in recent lazer builds, any screen with an automatically-focused text input, such as the song list or the settings panel, will open the on-screen keyboard. This requires repeatedly closing the keyboard while the screen is open in order to navigate. Opening the keyboard does seem useful but probably only once I click on the text box, but I don't know how straightforward that autofocus behavior is to modify (maybe depending on the value of [`SDL_HasScreenKeyboardSupport`](https://wiki.libsdl.org/SDL2/SDL_HasScreenKeyboardSupport)?)

This started in `2022.1227.0`, and didn't happen in `2022.1208.0`. It looks like this behavior was added to SDL's text input with https://github.com/libsdl-org/SDL/pull/6515 and then `2022.1227.0` got it via updating SDL.

### Screenshots or videos

https://user-images.githubusercontent.com/1427698/212415317-f4032fa3-32ea-4c70-a2b7-29d8ef89a6dc.mp4

### Version

2022.1228.0-lazer

### Logs

[database.log](https://github.com/ppy/osu/files/10415313/database.log)
[input.log](https://github.com/ppy/osu/files/10415314/input.log)
[network.log](https://github.com/ppy/osu/files/10415315/network.log)
[performance.log](https://github.com/ppy/osu/files/10415316/performance.log)
[runtime.log](https://github.com/ppy/osu/files/10415317/runtime.log)

**Comment by bdach** [comment]
The text boxes actually do not autofocus on mobile platforms, due to this:

https://github.com/ppy/osu/blob/ad2582a3ab8d4da2497bb4fa559bbe22246cd3a7/osu.Game/Graphics/UserInterface/FocusedTextBox.cs#L22

`OnScreenKeyboardOverlapsGameWindow` is false for mobile game hosts, and is hardcoded to true on desktop platforms. Steam Deck is in the gray area and is presumably reporting the latter. `SDL_HasScreenKeyboardSupport` may be the correct thing to apply here, but care needs to be taken to verify that it doesn't also return true on normal windows/linux/macOS too, defeating the point of the entire exercise.

Relatively low priority issue though, especially though that reproduction requires owning the hardware (which I do not).

**Comment by Susko3** [comment]
> May also be relevant on surface tablets or similar touch-based-but-desktop devices.

`SDL_HasScreenKeyboardSupport` returns false on touch-supported Windows systems. And indeed, the on-screen keyboard doesn't open when `SDL_StartTextInput` is called.

> SDL_HasScreenKeyboardSupport may be the correct thing to apply here, but care needs to be taken to verify that it doesn't also return true on normal windows/linux/macOS too, defeating the point of the entire exercise.

`SDL_HasScreenKeyboardSupport` doesn't quite do what we need. For example, if you plug in a keyboard to the Deck, you'd expect it to behave as it currently is -- focusing the text box. But `SDL_HasScreenKeyboardSupport` will still return `true` in that case (looking at the SDL code), so applying it as-is won't work.

**Comment by NatParable** [comment]
I just tried the latest version on my Steam Deck and this is still happening. Is there no way to work around this? It makes it almost impossible to go through the song list or the options menu as the on-screen keyboard keeps popping up every time I touch the screen.

**Comment by sciboy12** [comment]
Try adding `SDL_ENABLE_SCREEN_KEYBOARD=0` as an environment variable to osu!'s launch options under Steam.
I've found that this works for me, and the OSK can still be accessed by pressing Steam + X.

**Comment by NatParable** [comment]
Thank you! That works! 
For anyone having trouble getting sciboy12's advice to work, you have to add "%command%" (no quotation marks) after the sdl variable in launch parameters.
